### DISCUSSION TITLE
[5.x] FR: More convenient way to access custom fields in JSON column for advanced queries

### DISCUSSION THREAD

**Discussion by wsydney76** [craftcms/cms#14557] [Ideas] [answered]
How to use custom fields in advanced queries is described [here](https://craftcms.com/docs/5.x/upgrade.html#advanced-queries).

Had to do this a lot today when upgrading a project with quite complex queries, (with the help of a custom twig extension). 

Would be great though to have an easier way in core, something like `valueSql('entryTypeHandle', 'fieldHandle').

So that you can for example do this instead of manually going through all the documented steps:

```twig
{% set lastSeason = filmQuery.max(valueSql('film', 'season')) %}
```
and a corresponding PHP function `SomeHelper::getValueSql('film', 'season')`

**Comment by brandonkelly** [comment, marked as answer]
Just added a new `fieldValueSql()` Twig function for the next release.

Custom fields can be added to more than just entry types (volumes, global sets, the user field layout, etc.), so the first argument accepts a `FieldLayoutProviderInterface` instance (e.g. `EntryType`) rather than an entry type handle, and I also added a new `entryType()` method to fetch the entry type:

```twig
{% set lastSeason = filmQuery.max(fieldValueSql(entryType('film'), 'season')) %}
```

(7c9746527e898c68118f8a07eb5c6d211803576c)

**Reply by brandonkelly** [reply]
5.0.0-beta.7 is out now with those new functions.
